analysers/analyser_merge_afigeo_hydrants.py

```diff
@@ -0,0 +1,92 @@
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+from .Analyser_Merge import Analyser_Merge, Source, GeoJSON, Load, Mapping, Select, Generate
+
+
+class _Analyser_Merge_Afigeo_Hydrants(Analyser_Merge):
+    def __init__(self, config, source_url, dataset_name, source, osmRef, logger = None):
+        Analyser_Merge.__init__(self, config, logger)
+        # TODO: no idea how we pick the item number, and the id and level
+        self.def_class_missing_official(item = 8360, id = 1, level = 3, tags = ['merge', 'emergency'],
+            title = T_('Fire hydrant not integrated'))
+        self.def_class_possible_merge(item = 8361, id = 3, level = 3, tags = ['merge', 'emergency'],
+            title = T_('Fire hydrant integration suggestion'))
+        self.def_class_update_official(item = 8362, id = 4, level = 3, tags = ['merge', 'emergency'],
+            title = T_('Fire hydrant update'))
+
+        def extract_water_source(res):
+            if res.get('SOURCE_PEI') == 'piscine':
+                return 'swimming_pool'
+            elif res.get('SOURCE_PEI') == 'puits':
+                return 'well'
+            elif res.get('SOURCE_PEI') == 'reseau_irrigation':
+                return 'groundwater'
+            elif res.get('SOURCE_PEI') == 'citerne' or res.get('TYPE_PEI') == 'CI':
+                return 'water_tank'
+            elif res.get('SOURCE_PEI') == 'reseau_aep':
+                return 'main'
+            else:
+                return Generate.delete_tag
+
+        self.init(
+            source_url,
+            dataset_name,
+            GeoJSON(source,
+                extractor = lambda geojson: geojson),
+            Load("geom_x", "geom_y"),
+            Mapping(
+                select = Select(
+                    types = ["nodes"],
+                    tags = {"emergency": "fire_hydrant"}),
+                osmRef = osmRef,
+                conflationDistance = 50,
+                generate = Generate(
+                    static1 = {"source": self.source},
+                    mapping1 = {
+                        osmRef: "ID_SDIS",
+                        "emergency": lambda res: "fire_hydrant" if str(res.get('disponible')) != '0' else Generate.delete_tag,
+                        "disused:emergency": lambda res: "fire_hydrant" if str(res.get('disponible')) == '0' else Generate.delete_tag,
```

Comment:
Cannot map on `disused:emergency` since the conflation is done on `tags = {"emergency": "fire_hydrant"}`.